scripts/gen_single_image_diffusion.py

```diff
@@ -74,6 +78,38 @@ def to_np(img):
     return img
 
 
+def compute_mask_with_sam(img, rect_mask, sam_model, device):
```

Comment:
We may want this in `sam_inference.py` or `mask_generation.py` and import it here instead of having everything in the inference generic script.